library/ssl_tls13_client.c

```diff
@@ -2686,7 +2687,7 @@ static int ssl_encrypted_extensions_parse( mbedtls_ssl_context* ssl,
             case MBEDTLS_TLS_EXT_ALPN:
                 MBEDTLS_SSL_DEBUG_MSG( 3, ( "found alpn extension" ) );
 
-                if( ( ret = ssl_parse_alpn_ext( ssl, ext + 4, (size_t)ext_size ) ) != 0 )
+                if( ( ret = ssl_parse_alpn_ext( ssl, ext + 4, (size_t) ext_size ) ) != 0 )
                 {
                     return( ret );
                 }
```

Comment:
About the server name extension 'parsing' below:
* The comment suggests that the server name extension has to be empty. In this case, we would need to check for it and fail with a parsing error if it's not empty.
* However, [RFC 6066](https://tools.ietf.org/html/rfc6066) only says the extension SHOULD be empty, which suggests that it's safer to just ignore the contents here. 

Could you reflect this in the comment on L.2701?